Sources/WalletConnectSign/Services/DisconnectService.swift

```diff
@@ -1,20 +1,26 @@
 import Foundation
 
-class DisconnectService {
+final class DisconnectService {
+    
     enum Errors: Error {
         case sessionForTopicNotFound
     }
 
-    private let deleteSessionService: DeleteSessionService
+    private let deleteSessionService: DeleteSessionServiceProtocol
     private let sessionStorage: WCSessionStorage
+    private let invalidRequestsSanitiser: InvalidRequestsSanitiserProtocol
 
-    init(deleteSessionService: DeleteSessionService,
-         sessionStorage: WCSessionStorage) {
+    init(deleteSessionService: DeleteSessionServiceProtocol,
+         sessionStorage: WCSessionStorage,
+         invalidRequestsSanitiser: InvalidRequestsSanitiserProtocol) {
         self.deleteSessionService = deleteSessionService
         self.sessionStorage = sessionStorage
+        self.invalidRequestsSanitiser = invalidRequestsSanitiser
     }
 
     func disconnect(topic: String) async throws {
+        invalidRequestsSanitiser.removeSessionRequestsWith(topic: topic)
```

Comment:
A couple of notes:
* I'm not sure what the performance impact of this call is, if it blocks the thread and for how long
* Not sure if this is thread safe

We should check before merging.